projects/canopy/src/lib/modal/modal.component.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  EventEmitter,
+  forwardRef,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+
+import { Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { keyName } from '../utils/keyboard-keys';
+
+import { LgModalService } from './modal.service';
+import { LgModalHeaderComponent } from './modal-header/modal-header.component';
+import { LgModalBodyComponent } from './modal-body/modal-body.component';
+
+@Component({
+  selector: 'lg-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LgModalComponent implements OnInit, AfterContentInit, OnDestroy {
+  @Input() id: string;
+  @Input() closeOnOverlayClick = true;
+  @Output() open: EventEmitter<void> = new EventEmitter();
+  @Output() closed: EventEmitter<void> = new EventEmitter();
+  @ContentChild(forwardRef(() => LgModalHeaderComponent))
+  modalHeaderComponent: LgModalHeaderComponent;
+  @ContentChild(forwardRef(() => LgModalBodyComponent))
+  modalBodyComponent: LgModalBodyComponent;
+
+  isOpen: boolean;
+  subscriptions: Array<Subscription> = [];
```

Comment:
Just wondering about the array of subscriptions? it seems like there can only ever be one at the moment.

Reply:
Updated to use `subscription` 🙂